docs/config.rst

```diff
@@ -26,6 +26,11 @@ first override the same ones that may appear in the second one, and so on.
 .. tip::
   If you want to ignore any default configuration files and use only the custom one, use ``-C <config file>``.
 
+.. note::
+  Pipeline parameters (i.e. the ``params`` scope) should be overwritten through CLI options (``--something value``)
+  or a params file (``-params-file params.yml``). If you supply parameters through a custom config file (``-c my_config``),
+  it will be applied to the pipeline script but not to previous config files, such as the project's ``nextflow.config`` file.
+  See the :ref:`CLI page <cli>`. for more information about how to supply params through CLI options or a params file.
```

Comment:
Not understanding this

Reply:
This is in response to the related bug reported here: https://github.com/nextflow-io/nextflow/issues/2662

Where params in a custom config file supplied by `-c`are not actually sent to processes in the pipeline (even though e.g. tower reports they are correctly resolved).

I can try and rephrase to say more bluntly: use only param-files or cli `--` for specifying pipeline parameters. Don't use config files via `-c`. (DSL2 only).

But this seemed a bit extreme/negative

Reply:
@jfy133 I thought it was the other way around -- that the params are applied to the pipeline script but not the base config?

@pditommaso In any case I think the issue linked by James is a good test case that describes the problem. I think we should either add this note to the docs or fix the inconsistency in Nextflow if it isn't too complicated.

Reply:
Hm, ok yes the current description is flipped I think (I'll make a suggestion to clarify that) - sorry.

I don't know exactly what you mean via pipeline script vs. base config.  Basically what I found is with DSL2: 

if I provide a custom config that has the params scope, if I check `tower` (or whatever is displyed in the top of an nf-core run header), the resolved parameters correctly displays the updated parameters derived from the custom config.

But when you actually look in the `.command.sh` the command itself only has the settings defined in e.g. `base.config`. 

But I found writing this concisely a little difficult 💧

Reply:
I see. I think we need to find an explanation for this phenomenon in the source code before documenting it. It's a rather confusing edge case, and it might be better to fix it than to try to document it.

Reply:
Fair enough.

I could just set it as `Pipeline parameters (i.e. the ``params`` scope) should be overwritten through CLI options (``--something value``)
  or a params file (``-params-file params.yml``).`
  
  Or should I remove the note from this page entirely? The `params-file` descriptions would still be a useful addition regardless I think

Reply:
Sorry, still finding confusing the "previous configuration" concept. I've remove this paragraph